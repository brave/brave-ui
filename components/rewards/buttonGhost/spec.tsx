/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import ButtonGhost from './index'

describe('ButtonGhost tests', () => {
  const baseComponent = (props?: object) => <ButtonGhost id='button' {...props} />

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#button').length
      expect(assertion).toBe(1)
    })
  })
})
