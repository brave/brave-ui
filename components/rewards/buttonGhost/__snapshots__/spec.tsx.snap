// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGhost tests basic tests matches the snapshot 1`] = `
.c0 {
  display: inline-block;
  text-align: center;
  box-sizing: border-box;
  font-family: Poppins;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  font-weight: 600;
  line-height: 1;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: rgba(,0.9);
}

.c0:hover {
  color: rgba(,1);
}

.c0:active {
  color: rgba(,0.6);
}

<div
  className="c0"
  id="button"
/>
`;
